build_tools/cmake/iree_static_linker_test.cmake

```diff
@@ -0,0 +1,295 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# iree_static_linker_test()
+#
+# Creates a test test for the statically linked libraries generated by the
+# llvm-cpu compiler target backend and executed using the local-sync runtime
+# HAL driver.
+#
+# Parameters:
+#   NAME: Name of the target
+#   SRC: mlir source file to be compiled to an IREE module.
+#   INPUT_TYPE: Module input type, assuming native c/c++ data types
+#   HAL_TYPE: HAL data type, see runtime/src/iree/hal/buffer_view.h for possible
+#       options.
+#   STATIC_LIB_PREFIX: llvm static library prefix.
+#   ENTRY_FUNCTION: Entry function call.
+#   EMITC: Uses EmitC to output C code instead of VM bytecode.
+#   COMPILER_FLAGS: additional flags to pass to the compiler. Bytecode output
+#       format and backend flags are passed automatically.
+#   FUNCTION_INPUTS: Module input in the format of mxnxcx<datatype>.
+#   LABELS: Additional labels to apply to the test. The package path and
+#       "driver=local-sync" are added automatically.
+#   TARGET_CPU_FEATURES: If specified, a string passed as argument to
+#       --iree-llvm-target-cpu-features.
+#
+# Example:
+#   iree_static_linker_test(
+#     NAME
+#       edge_detection_test
+#     SRC
+#       "edge_detection.mlir"
+#     STATIC_LIB_PREFIX
+#       edge_detection_linked_llvm_cpu
+#     ENTRY_FUNCTION
+#       "edge_detect_sobel_operator"
+#     FUNCTION_INPUTS
+#       "1x128x128xf32"
+#     COMPILER_FLAGS
+#       "--iree-input-type=mhlo"
+#   )
+function(iree_static_linker_test)
+  if(NOT IREE_BUILD_TESTS)
+    return()
+  endif()
+
+  # See comment in iree_check_test about this condition.
+  if(NOT IREE_BUILD_COMPILER AND NOT CMAKE_CROSSCOMPILING)
+    return()
+  endif()
+
+  if(NOT IREE_TARGET_BACKEND_LLVM_CPU OR NOT IREE_HAL_DRIVER_LOCAL_SYNC)
+    return()
+  endif()
+
+  cmake_parse_arguments(
+    _RULE
+    "EMITC"
```

Comment:
Can you also add a check for `IREE_OUTPUT_FORMAT_C` after this?

https://github.com/iree-org/iree/blob/57ec69da106a2003ec5f383cbd72dcaed1d84fea/CMakeLists.txt#L235-L239

```cmake
  if(_RULE_EMITC AND NOT IREE_OUTPUT_FORMAT_C)
    return()
  endif()
```

Reply:
Done.